File: ai25btech11007/MatGeo/MatGeo_6/Codes/main.c
#include <stdio.h>
#include <math.h>

// Custom sqrt (overrides math.h sqrt)
double sqrt(double n) {
    if (n < 0) return -1; // invalid for negative numbers
    if (n == 0) return 0;

    double guess = n / 2.0;
    for (int i = 0; i < 20; i++) { // Newton-Raphson iterations
        guess = 0.5 * (guess + n / guess);
    }
    return guess;
}

// Function to calculate dot product of two vectors
double dotProduct(double A[], double B[]) {
    return A[0]*B[0] + A[1]*B[1];
}

// Function to calculate magnitude of a vector
double magnitude(double V[]) {
    return sqrt(V[0]*V[0] + V[1]*V[1]);
}

#include <stdio.h>
#include <math.h>

int main(void) {
    /* Given data */
    double a = 5.0;       /* BC */
    double K = 7.5;       /* AC + AB = b + c */
    double cosB = 0.5;    /* cos 60° */
    double sinB = sqrt(3.0) / 2.0; /* sin 60° */

    /* Formula from your solution */
    double c = (K*K - a*a) / (2.0 * (K - a * cosB));
    double b = K - c;

    /* Coordinates with B=(0,0), C=(a,0) */
    double Ax = (c * cosB) / sinB;
    double Ay = c / sinB;

    printf("Computed values:\n");
    printf("c (AC) = %.6f\n", c);
    printf("b (AB) = %.6f\n", b);
    printf("A = (%.6f, %.6f)\n", Ax, Ay);
    printf("B = (0.000000, 0.000000)\n");
    printf("C = (%.6f, 0.000000)\n", a);

    return 0;
}

File: ai25btech11007/MatGeo/MatGeo_6/Codes/main.py
import math
import matplotlib.pyplot as plt

# Given data
a = 5.0        # BC
K = 7.5        # AB + AC = b + c
cosB = 0.5     # cos(60°)
sinB = math.sqrt(3) / 2  # sin(60°)

# Formula from solution
c = (K**2 - a**2) / (2 * (K - a * cosB))
b = K - c

# Coordinates with B=(0,0), C=(a,0)
Ax = (c * cosB) / sinB
Ay = c / sinB
Bx, By = 0.0, 0.0
Cx, Cy = a, 0.0

print("Computed values:")
print(f"c (AC) = {c:.6f}")
print(f"b (AB) = {b:.6f}")
print(f"A = ({Ax:.6f}, {Ay:.6f})")
print(f"B = ({Bx:.6f}, {By:.6f})")
print(f"C = ({Cx:.6f}, {Cy:.6f})")

# Plotting the triangle
fig, ax = plt.subplots()
triangle_x = [Ax, Bx, Cx, Ax]
triangle_y = [Ay, By, Cy, Ay]
ax.plot(triangle_x, triangle_y, 'b-', linewidth=2)

# Mark points
ax.scatter([Ax, Bx, Cx], [Ay, By, Cy], color='red', zorder=5)
ax.text(Ax, Ay+0.2, "A", fontsize=12, ha='center')
ax.text(Bx, By-0.2, "B", fontsize=12, ha='center')
ax.text(Cx, Cy-0.2, "C", fontsize=12, ha='center')

# Formatting
ax.set_aspect('equal')
ax.grid(True, linestyle='--', alpha=0.6)
ax.set_title("Construction of Triangle ABC")

plt.show()
